### ISSUE TITLE
[Component] Chip: Add selected state. Remove Filled variant.

### ISSUE THREAD

**Issue by michael-zimmermann** [adevinta/spark-ios#469]

**Comment by elisa-hery** [comment]
reminder for testing : don't forget to test these comments --> https://github.com/adevinta/spark-ios/issues/457#issuecomment-1759760009

**Comment by elisa-hery** [comment]
@michael-zimmermann I tested chips, here are my comments : 
- all my [previous comments](https://github.com/adevinta/spark-ios/issues/457#issuecomment-1759760009) have been fixed ✅  except for the 3rd one : for alert intent, color of the text does not seem good
- on swiftUI, the disabled state is not working 
https://github.com/adevinta/spark-ios/assets/142396647/51632c68-3681-47a9-a4ba-ea3f6b443ab7
- surface button : for both swift UI & UIkit, hover background seems really blue, is it normal ?
https://github.com/adevinta/spark-ios/assets/142396647/76de9cd9-64d8-471a-aded-7766ffdab126
- surface button : for swift UI button does not have background so I cannot really check it. For dashed/selected I cannot see the dashed border.  

ℹ for other people : on iOS, the chip component itself is not responsible for setting the selected state, the use will have to do this. So on demo app there is no way to actually select a chip by pressing on it.

**Comment by michael-zimmermann** [comment]
1. I've fixed the alert intent text color.
2. I've the disabled state on SwiftUI
3. The surface intent pressed colors that we have in SparkCore are as below (Hex #E0EAFF). It is different from the Figma spec (background: #F2F6FF). To change this, we'll need to update our color assets.
<img width="972" alt="Screenshot 2023-10-18 at 17 02 06" src="https://github.com/adevinta/spark-ios/assets/128726463/022ed42f-5c84-40e2-a563-6d9619944a34">
4. I've updated background of the chip when the surface intent is selected to be able to see the border and text.
